### PR TITLE
feat: add a method for calling python APIs

### PR DESCRIPTION
**Pull request by Faraz32123**
- add a method named `check_service_enabled` for checking if the forum service
is enabled or not as it was being checked in v1
- call python methods instead of HTTPs APIs for getting response/data
for pin, unpin thread and get user's data by user_id
- add a method named `handle_response` for handling returned results from
forum v2 methods as same as v1 responses were handled

### CODE REVIEW

File: openedx/core/djangoapps/django_comment_common/comment_client/utils.py [hunk 1]
```diff
@@ -204,3 +218,46 @@ def check_forum_heartbeat():
             return 'forum', False, res.get('check', 'Forum heartbeat failed')
     except Exception as fail:
         return 'forum', False, str(fail)
+
+
+def check_service_enabled():
```
Comment (regisb) [inline openedx/core/djangoapps/django_comment_common/comment_client/utils.py:223]: What is this for? The fact that we have to repeatedly call `check_service_enabled()` is a red flag, and we should get rid of that function.

File: openedx/core/djangoapps/django_comment_common/comment_client/utils.py [hunk 2]
```diff
@@ -204,3 +218,46 @@ def check_forum_heartbeat():
...
+    # To avoid dependency conflict
+    from openedx.core.djangoapps.django_comment_common.models import ForumsConfig
+
+    config = ForumsConfig.current()
+
+    if not config.enabled:
+        raise CommentClientMaintenanceError("service disabled")
+
+
+def handle_response(
```
Comment (regisb) [inline openedx/core/djangoapps/django_comment_common/comment_client/utils.py:233]: This function is preventing us from benefiting some of the niceties of a native API: explicit function signature, type checking, etc.

Instead, I propose that the forum API does not return status codes, but raises exceptions directly. Also, we should get rid of `metric_action` and raw requests.

**Review by regisb — CHANGES_REQUESTED** [review]
We need to make this API more native. Calling the forum API should be as easy as calling a regular function from the forum.api module.

File: openedx/core/djangoapps/django_comment_common/comment_client/utils.py (hunk 1 above)
Comment (Faraz32123) [reply openedx/core/djangoapps/django_comment_common/comment_client/utils.py:223]: I think we can remove this function/code as we have already added `ENABLE_DISCUSSION_SERVICE=true` in our forum_v2 plugin.

**Comment by Faraz32123** [comment]
closing this PR in favor of this [PR](https://github.com/openedx/edx-platform/pull/35490)

**Pull request closed without merging**